apps/openmw/mwmechanics/mechanicsmanagerimp.cpp

```diff
@@ -992,14 +992,19 @@ namespace MWMechanics
         }
     }
 
-    bool MechanicsManager::isItemStolenFrom(const std::string &itemid, const std::string &ownerid)
+    bool MechanicsManager::isItemStolenFrom(const std::string &itemid, const MWWorld::Ptr& ptr)
     {
         StolenItemsMap::const_iterator it = mStolenItems.find(Misc::StringUtils::lowerCase(itemid));
         if (it == mStolenItems.end())
             return false;
+
         const OwnerMap& owners = it->second;
+        const std::string ownerid = ptr.getCellRef().getRefId();
+        const std::string factionid = ptr.getClass().getPrimaryFaction(ptr);
```

Comment:
What will happen if the trader has no faction? Just unnecessary search for empty string?

Reply:
Ouch. Thanks.